src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(stateClone, obj.extraData);
+        stateHistory.push({ ...stateClone });
+        break;
+
+      case 'removeProperties':
+        for (const item of obj.keysToRemove) {
+          delete stateClone[item];
+        }
+        stateHistory.push({ ...stateClone });
```

Comment:
The state history update should occur outside of the `switch` block, at the end of each loop cycle. This ensures consistency in how state changes are recorded. Consider moving `stateHistory.push({ ...stateClone });` to after the `switch` block.